app/controllers/your-psp/post-flex.controller.test.js

```diff
@@ -0,0 +1,93 @@
+const sinon = require('sinon')
+
+const gatewayAccountFixtures = require('../../../test/fixtures/gateway-account.fixtures')
+
+describe('Post 3DS Flex controller', () => {
+  const gatewayAccountExternalId = 'a-gateway-account-external-id'
+  const credentialId = 'a-valid-credential-id'
+  let req
+  let res
+  let next
+  let postCheckWorldpay3dsFlexCredentials
+  let post3dsFlexAccountCredentials 
+  let updateIntegrationVersion3dsMock
+  let renderErrorViewMock
+
+  beforeEach(() => {
+    req = {
+      params: { credentialId },
+      flash: sinon.spy(),
+      body: {
+        'organisational-unit-id': '111111111111111111111111',
+        'issuer': '111111111111111111111111',
+        'jwt-mac-key': '11111111-1111-1111-1111-111111111111'
+      }
+    }
+    res = {
+      setHeader: sinon.stub(),
+      status: sinon.spy(),
+      redirect: sinon.spy(),
+      render: sinon.spy()
+    }
+    next = sinon.spy()
+  })
+
+  it('when `account.type=live`, then should toggle 3DS Flex on by setting 3DS integration version to 2', async () => {
+    req.account = getGatewayAcountWithType('live')
+    
+    postCheckWorldpay3dsFlexCredentials = sinon.spy(() => Promise.resolve({result: 'valid'}))
+    post3dsFlexAccountCredentials = sinon.spy(() => Promise.resolve())
+    updateIntegrationVersion3dsMock = sinon.spy(() => Promise.resolve())
+    const controller = getControllerWithMocks()
+
+    await controller(req, res, next)
+
+    sinon.assert.calledWith(updateIntegrationVersion3dsMock, req.account.gateway_account_id, 2)
+    sinon.assert.calledWith(req.flash, 'generic', 'Your Worldpay 3DS Flex settings have been updated')
+    sinon.assert.calledWith(res.redirect, `/account/${gatewayAccountExternalId}/your-psp/${credentialId}`)
+  })
+
+  it('when `account.type=test`, then it should NOT toggle 3DS Flex on by setting 3DS integration version to 2', async () => {
+    req.account = getGatewayAcountWithType('test')
+    
+    postCheckWorldpay3dsFlexCredentials = sinon.spy(() => Promise.resolve({result: 'valid'}))
+    post3dsFlexAccountCredentials = sinon.spy(() => Promise.resolve())
+    updateIntegrationVersion3dsMock = sinon.spy(() => Promise.resolve())
+    
+    const controller = getControllerWithMocks()
+
+    await controller(req, res, next)
+
+    sinon.assert.notCalled(updateIntegrationVersion3dsMock)
+  }) 
+
+  function getControllerWithMocks () {
+    return proxyquire('./post-flex.controller', {
+      '../../services/clients/connector.client': {
+        ConnectorClient: function () {
+          this.postCheckWorldpay3dsFlexCredentials = postCheckWorldpay3dsFlexCredentials
+          this.post3dsFlexAccountCredentials = post3dsFlexAccountCredentials
+          this.updateIntegrationVersion3ds = updateIntegrationVersion3dsMock
+        }
+      },
+      '../../utils/response': {
+        renderErrorView: renderErrorViewMock
+      }
+    })
+  }
+
+  function getGatewayAcountWithType(accountType = 'test') {
```

Comment:
Any reason a default `test` is required here?

Reply:
Good spot - have removed that